web/client/components/buttons/ZoomToMaxExtentButton.jsx

```diff
@@ -0,0 +1,100 @@
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { connect } from 'react-redux';
+import { changeMapView } from '../../actions/map';
+
+var React = require('react');
+var BootstrapReact = require('react-bootstrap');
+var Button = BootstrapReact.Button;
+var Glyphicon = BootstrapReact.Glyphicon;
+
+
+/**
+ * A button to zoom to max. extent of the map or zoom level one.
+ * Component's properies:
+ *  - id: {string}            custom identifier for this component
+ *  - style: {object}         a css-like hash to define the style on the component
+ *  - glyphicon: {string}     bootstrap glyphicon name
+ *  - text: {string|element}  text content for the button
+ *  - btnSize: {string}       bootstrap button size ('large', 'small', 'xsmall')
+ */
+var ZoomToMaxExtentButton = React.createClass({
+    propTypes: {
+        id: React.PropTypes.string,
+        style: React.PropTypes.object,
+        glyphicon: React.PropTypes.string,
+        text: React.PropTypes.string,
+        btnSize: React.PropTypes.oneOf(['large', 'medium', 'small', 'xsmall']),
+        // redux store slice with map configuration (bound through connect to store at the end of the file)
+        mapConfig: React.PropTypes.object,
+        // redux store dispatch func
+        dispatch: React.PropTypes.func
+    },
+    getDefaultProps() {
+        return {
+            id: "mapstore-zoomtomaxextent",
+            style: undefined,
+            glyphicon: "resize-full",
+            text: undefined,
+            hiddenText: true,
+            btnSize: 'xsmall'
+        };
+    },
+    render() {
+        return (
+            <Button
+                id={this.props.id}
+                bsStyle="default"
+                bsSize={this.props.btnSize}
+                onClick={() => this.zoomToMaxExtent()}>
+                {this.props.glyphicon ? <Glyphicon glyph={this.props.glyphicon}/> : null}
+                {this.props.glyphicon && this.props.text ? "\u00A0" : null}
+                {this.props.text}
+            </Button>
+        );
+    },
+    zoomToMaxExtent() {
+        var mapConfig = this.props.mapConfig;
+        var maxExtent = mapConfig.maxExtent;
+        var bbox;
+
+        if (maxExtent &&
+            Object.prototype.toString.call(maxExtent) === '[object Array]') {
+            // zoom map to the max. extent defined in the map's config
+            bbox = {
+                bounds: {
+                    minx: maxExtent[0],
+                    miny: maxExtent[1],
+                    maxx: maxExtent[2],
+                    maxy: maxExtent[3]
+                },
+                crs: mapConfig.projection,
+                rotation: 0
+            };
+            // adapt the map view by dispatching to the corresponding action
+            this.props.dispatch(changeMapView(this.props.mapConfig.center, -1,
```

Comment:
You should get the action to call as a property without directly calling dispatch